read.go

```diff
@@ -50,14 +50,6 @@ func (r *reader) readTag() (*tag.Tag, error) {
 	if gerr == nil && eerr == nil {
 		return &tag.Tag{Group: group, Element: element}, nil
 	}
-	// If the error is an io.EOF, return the error directly instead of the compound error later.
```

Comment:
Why are these lines removed? (This is probably causing the test failures).

Reply:
That was not intentional.  Don't even remember touching this section but I have restored it.